src/DocumentDB.ChangeFeedProcessor.IntegrationTests/Utils/IntegrationTestsHelper.cs

```diff
@@ -77,7 +80,19 @@ internal static async Task CreateDocumentCollectionAsync(DocumentClient client,
             var database = new Database { Id = databaseId };
             database = await client.CreateDatabaseIfNotExistsAsync(database);
 
-            await client.CreateDocumentCollectionAsync(database.SelfLink, collection, new RequestOptions { OfferThroughput = offerThroughput });
+            int retryCount = 3;
+            while (retryCount-- > 0)
+            {
+                try
+                {
+                    await client.CreateDocumentCollectionAsync(database.SelfLink, collection, new RequestOptions { OfferThroughput = offerThroughput });
+                    break;
+                }
+                catch (DocumentClientException)
+                {
+                    // Public emulator might have transient
```

Comment:
when all retries are over should we throw?

Reply:
Good catch